kernel/libc/koslib/pipe.c

```diff
@@ -0,0 +1,37 @@
+/* KallistiOS ##version##
+
+   pipe.c
+   Copyright (C) 2024 Andress Barajas
+
+*/
+
+#include <kos/fs_pty.h>
+#include <kos/fs.h>
+#include <errno.h>
+
+int pipe(int pipefd[2]) {
+    int master_fd, slave_fd;
+
+    if(pipefd == NULL) {
+        errno = EFAULT;
+        return -1;
+    }
+
+    /* Create a PTY master/slave pair */
+    if(fs_pty_create(NULL, 0, &master_fd, &slave_fd) < 0)
+        return -1;
+
+    if(master_fd < 0)
+        return -1;
+
+    if(slave_fd < 0) {
+        fs_close(master_fd);
+        return -1;
+    }
```

Comment:
Sorry for the late stray review, I overlooked pipe the first time around.

I think, possibly, that this checking represents a gap in `fs_pty_create`s error handling. If either of those fds aren't valid, then doesn't that mean the pty creation failed? Or is that behavior allowed for some reason?

If that's not an issue, then this should try to close both, as master_fd being invalid doesn't mean that slave_fd is also invalid

Reply:
Youre right.  Looks to be a gap in fs_pty_creates error handling.  I'll take a look tomorrow.

Reply:
@QuzarDC I updated pipe() to count on fs_pty_create() to do the right error handling.  Updated fs_pty_create() to handle cleanup if the master or slave files fail to open.